geomesa-native-api/src/main/java/org/locationtech/geomesa/api/ValueSerializer.java

```diff
@@ -0,0 +1,13 @@
+package org.locationtech.geomesa.api;
+
+/**
+ * Required by GeoMesaIndex in order to serialize
+ * a value object into the index
+ * @param <T>
+ */
+public interface ValueSerializer<T> {
```

Comment:
later on down the road might be nice to provide ```reuse``` objects in the api but this is OK for now